drivers/clock_control/clock_control_nrf2_common.c

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "clock_control_nrf2_common.h"
+LOG_MODULE_REGISTER(clock_control_nrf2);
+
+#define FLAG_UPDATE_IN_PROGRESS BIT(FLAGS_COMMON_BITS - 1)
+#define FLAG_UPDATE_NEEDED      BIT(FLAGS_COMMON_BITS - 2)
+
+#define ONOFF_CNT_MAX (FLAGS_COMMON_BITS - 2)
+
+#define CONTAINER_OF_ITEM(ptr, idx, type, array) \
+	(type *)((char *)ptr - \
+		 (idx * sizeof(array[0])) - \
+		 offsetof(type, array[0]))
+
+/*
+ * Definition of `struct clock_config_generic`.
+ * Used to access `clock_config_*` structures in a common way.
+ */
+STRUCT_CLOCK_CONFIG(generic, ONOFF_CNT_MAX);
+
+static void update_config(struct clock_config_generic *cfg)
+{
+	atomic_val_t prev_flags = atomic_or(&cfg->flags, FLAG_UPDATE_NEEDED);
+
+	if (prev_flags & (FLAG_UPDATE_NEEDED | FLAG_UPDATE_IN_PROGRESS)) {
```

Comment:
I'm confused here. It look like line 27 results in `prev_flags` equal to `FLAG_UPDATE_NEEDED` or `0`. So what is the purpose of checking `FLAG_UPDATE_IN_PROGRESS` here?

Reply:
> I'm confused here. It look like line 27 results in `prev_flags` equal to `FLAG_UPDATE_NEEDED` or `0`.

No, `prev_flags` will have `FLAG_UPDATE_IN_PROGRESS` set if the update work is currently executed, and it will have `FLAG_UPDATE_NEEDED` if an update is already scheduled. In both these cases, the update work is not to be submitted (in the former case, it will be submitted at the end of `clock_config_update_end()`).

Reply:
I see. Too many reviews in a single day :)